test/limitOrderBook/limitOrderBook.fillLimitOrder.triggerPrice.ts

```diff
@@ -0,0 +1,457 @@
+                oppositeAmountBound: parseEther("300").toString(),
+                deadline: ethers.constants.MaxUint256.toString(),
+                sqrtPriceLimitX96: 0,
+                referralCode: ethers.constants.HashZero,
+                reduceOnly: false,
+                roundIdWhenCreated: "0",
+                triggerPrice: "0",
+            }
+
+            const signature = await getSignature(fixture, stopLimitOrder, trader)
+
+            await expect(
+                limitOrderBook.connect(keeper).fillLimitOrder(stopLimitOrder, signature, computeRoundId(1, 1)),
+            ).to.revertedWith("LOB_IRI")
+        })
+
+        it("force error, roundIdWhenTriggered is earlier than roundIdWhenCreated", async () => {
+            const stopLimitOrder = {
+                orderType: fixture.orderTypeStopLimitOrder,
+                salt: 1,
+                trader: trader.address,
+                baseToken: baseToken.address,
+                isBaseToQuote: false,
+                isExactInput: false,
+                amount: parseEther("0.1").toString(),
+                oppositeAmountBound: parseEther("300").toString(),
+                deadline: ethers.constants.MaxUint256.toString(),
+                sqrtPriceLimitX96: 0,
+                referralCode: ethers.constants.HashZero,
+                reduceOnly: false,
+                roundIdWhenCreated: computeRoundId(1, 2),
+                triggerPrice: "0",
+            }
+
+            const signature = await getSignature(fixture, stopLimitOrder, trader)
+
+            await expect(
+                limitOrderBook.connect(keeper).fillLimitOrder(stopLimitOrder, signature, computeRoundId(1, 1)),
+            ).to.revertedWith("LOB_IRI")
+        })
+
+        it("force error, triggerPrice is 0", async () => {
+            const stopLimitOrder = {
+                orderType: fixture.orderTypeStopLimitOrder,
+                salt: 1,
+                trader: trader.address,
+                baseToken: baseToken.address,
+                isBaseToQuote: false,
+                isExactInput: false,
+                amount: parseEther("0.1").toString(),
+                oppositeAmountBound: parseEther("300").toString(),
+                deadline: ethers.constants.MaxUint256.toString(),
+                sqrtPriceLimitX96: 0,
+                referralCode: ethers.constants.HashZero,
+                reduceOnly: false,
+                roundIdWhenCreated: computeRoundId(1, 1),
+                triggerPrice: "0",
+            }
+
+            const signature = await getSignature(fixture, stopLimitOrder, trader)
+
+            await expect(
+                limitOrderBook.connect(keeper).fillLimitOrder(stopLimitOrder, signature, computeRoundId(1, 2)),
+            ).to.revertedWith("LOB_ITP")
+        })
+
+        // TODO
+        it("force error, roundId isn't existed", async () => {})
+        it("force error, baseToken isn't using ChainlinkPriceFeed", async () => {})
```

Comment:
This one seems pretty important and we'll have to do it; perhaps at least knowing what error message will be emitted and it won't change each time!

Reply:
Yes, added! Please review https://github.com/perpetual-protocol/perp-curie-periphery-contract/pull/14/commits/e6dd162c2be00b83cfcdd8cdf6b07f1b38350f1a

Reply:
No I haven't haha doing it now

Reply:
Done; great job!